src/extractor/edge_based_graph_factory.cpp

```diff
@@ -416,56 +425,79 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
                 BOOST_ASSERT(!edge_data2.reversed);
 
                 // the following is the core of the loop.
-                unsigned distance = edge_data1.distance;
+                unsigned weight = edge_data1.weight;
+                unsigned duration = edge_data1.duration;
                 if (m_traffic_lights.find(node_v) != m_traffic_lights.end())
                 {
-                    distance += profile_properties.traffic_signal_penalty;
+                    weight += profile_properties.traffic_signal_penalty;
                 }
 
-                const int32_t turn_penalty =
-                    scripting_environment.GetTurnPenalty(180. - turn.angle);
-                const auto turn_instruction = turn.instruction;
-
-                if (turn_instruction.direction_modifier == guidance::DirectionModifier::UTurn)
+                ExtractionTurn extracted_turn(guidance::isUturn(turn.instruction), 180. - input_turn.angle);
+                scripting_environment.ProcessTurn(extracted_turn);
+                if (fallback_to_duration && extracted_turn.weight < 0 && extracted_turn.duration > 0)
                 {
-                    distance += profile_properties.u_turn_penalty;
+                    turn.weight = turn.duration * 10.; // convert from seconds in deci-seconds
                 }
+                weight += std::max(0, extracted_turn.weight);
+                duration += std::max(0, extracted_turn.duration);
 
-                distance += turn_penalty;
-
+                // First write out the meta information of a turn edge
                 BOOST_ASSERT(m_compressed_edge_container.HasEntryForID(edge_from_u));
                 original_edge_data_vector.emplace_back(
                     m_compressed_edge_container.GetPositionForID(edge_from_u),
                     edge_data1.name_id,
                     turn.lane_data_id,
-                    turn_instruction,
+                    turn.instruction,
                     entry_class_id,
                     edge_data1.travel_mode);
-
                 ++original_edges_counter;
-
                 if (original_edge_data_vector.size() > 1024 * 1024 * 10)
                 {
                     FlushVectorToStream(edge_data_file, original_edge_data_vector);
                 }
 
+                // compute weight and duration penalties
+                std::uint16_t weight_penalty = 0;
+                std::uint16_t duration_penalty = 0;
+                if (m_traffic_lights.find(node_v) != m_traffic_lights.end())
+                {
+                    // FIXME we can't assume this here!
+                    weight_penalty += profile_properties.traffic_signal_penalty;
+                    duration_penalty += profile_properties.traffic_signal_penalty;
+                }
+
+                if (use_turn_function)
+                {
+                    const auto extracted_turn = ProcessTurn(turn, lua_state);
```

Comment:
`ProcessTurn(ExtractionTurn &turn) ` must be adjusted here, also `lua_state` is not defined